tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -354,6 +353,9 @@ def __call__(self, i):
             # if action_date is timezone aware, set to timezone provided
             action_date = action_date.astimezone(self.tz)
             self.current_date = now(tz=self.tz)
+        else:
+            # if it is not need to reset to timezone naive
```

Comment:
Good catch! I think we can get rid of `self.current_date` entirely and replace it with function-scoped `current_date`

Reply:
Good idea, that should really clarify things